scripts/oe-env-prepare.ps1

```diff
@@ -0,0 +1,446 @@
+    }
+    foreach($driver in $drivers.Keys) {
+        $zip = Get-Item $drivers[$driver]['zip_path']
+        if(!$zip) {
+            Throw "Cannot find the zile file with $driver"
+        }
+        if($zip.Count -gt 1) {
+            $zip
+            Throw "Multiple driver zip files found"
+        }
+        New-Item -ItemType Directory -Force -Path "$PACKAGES_DIRECTORY\Intel_SGX_DCAP\$driver"
+        Expand-Archive -Path $zip -DestinationPath "$PACKAGES_DIRECTORY\Intel_SGX_DCAP\$driver" -Force
+        $inf = Get-Item "$PACKAGES_DIRECTORY\Intel_SGX_DCAP\$driver\drivers\*\$driver.inf"
+        if(!$inf) {
+            Throw "Cannot find $driver.inf file"
+        }
+        if($inf.Count -gt 1) {
+            $inf
+            Throw "Multiple $driver.inf files found"
+        }
+        & "$PACKAGES_DIRECTORY\devcon.exe" install "$($inf.FullName)" $drivers[$driver]['location']
+        if($LASTEXITCODE) {
+            Throw "Failed to install $driver driver"
+        }
+    }
+    $OE_NUGET_DIR = "C:\openenclave\prereqs\nuget"
+    $TEMP_NUGET_DIR = "$PACKAGES_DIRECTORY\Azure_DCAP_Client_nupkg"
+    New-Directory -Path $OE_NUGET_DIR
+    New-Directory -Path $TEMP_NUGET_DIR
+    Copy-Item "$PACKAGES_DIRECTORY\Intel_SGX_DCAP\Intel SGX DCAP for Windows *\nupkg\*\*.nupkg" -Destination $TEMP_NUGET_DIR -Force
+    Copy-Item $PACKAGES['azure_dcap_client_nupkg']['local_file'] -Destination $TEMP_NUGET_DIR -Force
+    & "$PACKAGES_DIRECTORY\nuget.exe" install 'EnclaveCommonAPI' -Source "$TEMP_NUGET_DIR" -OutputDirectory "$OE_NUGET_DIR" -ExcludeVersion
+    & "$PACKAGES_DIRECTORY\nuget.exe" install 'DCAP_Components' -Source "$TEMP_NUGET_DIR" -OutputDirectory "$OE_NUGET_DIR" -ExcludeVersion
+    & "$PACKAGES_DIRECTORY\nuget.exe" install 'Microsoft.Azure.DCAP.Client' -Source "$TEMP_NUGET_DIR;nuget.org" -OutputDirectory "$OE_NUGET_DIR" -ExcludeVersion
+
+    New-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\sgx_lc_msr\Parameters" -Name "SGX_Launch_Config_Optin" -Value "1" -PropertyType DWORD -Force
+
+    Add-ToSystemPath -Path 'C:\Program Files\shellcheck'
+    Add-ToSystemPath -Path 'C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build'
+    Add-ToSystemPath -Path 'C:\Program Files\LLVM\bin'
+}
+
+try {
+    New-Directory -Path ${AZUREDATA_DIRECTORY}
+    New-Directory -Path ${AZUREDATA_BIN_DIRECTORY}
+
+    Start-LocalPackagesDownload
+    Install-Git
+    Install-7Zip
+
+    Write-Output "Installing Open Enclave"
+    Install-SGX
+    Install-PSW
+
+    Start-ExecuteWithRetry -ScriptBlock {
+        Start-Service "AESMService" -ErrorAction Stop
+    } -RetryMessage "Failed to start AESMService. Retrying"
+
+    Copy-Item -Path $PACKAGES["nuget"]["local_file"] -Destination "${AZUREDATA_BIN_DIRECTORY}\nuget.exe"
+
+    Install-VisualStudio
+    Install-Ocaml
+    Install-LLVM
+    Install-Shellcheck
+    Install-DCAPDrivers
+    Write-Output 'Please reboot your computer for the configuration to complete.'
```

Comment:
We should reboot the node at the end of the script (instead of `exit 0`).

And we should have the docs say that the script will reboot the node.

Reply:
Rebooting from the script makes ansible lose connection to the node:
`The error was: winrm.exceptions.WinRMTransportError: ('http', 'Bad HTTP response returned from server. Code 500')
`
I suggest using `win_reboot` from ansible and when run manually, instruct the user to reboot the node manually by displaying this message at the end of the script.

Reply:
Hum, you're right. But I think we can still reboot with the script, and wait with an Ansible task until the node becomes reachable again.

Keep in mind that I found out recently that `win_reboot` from Ansible doesn't properly reboot a SGX Windows VM from Azure. The node never comes back online, and you have to manually reboot it via the Azure API.

I am preparing a new pull request to have this fixed. Basically, we reboot the node via a CMD shell command, and we wait the node to become reachable over WinRM.

I suggest that you adopt the same approach here. Execute the script, which does a reboot at the end, and in the following task just wait for the node to become reachable over WinRM.

See my changes: https://github.com/ionutbalutoiu/openenclave/commit/96a56dc8c9df8a835d38c3c24865eab4a4c7e44f.

Reply:
Small update here.

Just synced offline with @george-caluta. It turns out that rebooting from the PowerShell script causes Ansible to sometimes error out the Ansible tasks that executes the PowerShell script:
```
TASK [Run the OE setup script] *******************************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: winrm.exceptions.WinRMTransportError: ('http', 'Bad HTTP response returned from server. Code 500')
fatal: [192.168.88.224]: FAILED! => {"msg": "Unexpected failure during module execution.", "stdout": ""}
```

Thus, we will be rebooting the machine outside the PowerShell script, and the script will only instruct that a reboot is needed.